src/crypto/algorithms/megolm.ts

```diff
@@ -1390,6 +1391,7 @@ class MegolmDecryption extends DecryptionAlgorithm {
         if (!senderPendingEvents.has(sessionId)) {
             senderPendingEvents.set(sessionId, new Set());
         }
+        // @ts-ignore: TS isn't smart enough to figure out `has` + `set` above makes this non-null
```

Comment:
sad :(

Reply:
AFAIK there is also no way to;
- have something like `setdefault`
- have TS recognise this in the first place with any kind of contract annotation

I could also do something funky like; (pseudocode)
```ts
const set = senderPendingEvents.get(sessionId) ?? { const newSet = new Set(); senderPendingEvents.set(sessionId, new Set()); return newSet }
```

Reply:
> have something like setdefault

there is by extending Map like we do in react-sdk but lets avoid that here

Reply:
Non-null assertions arent allowed, but we can do `?.` instead.